src/benchmarks.rs

```diff
@@ -0,0 +1,408 @@
+		max_pov_size: 1000u32,
+		parent_head: HeadData(vec![1, 1, 1]),
+	};
+	let inherent_data = {
+		let mut inherent_data = InherentData::default();
+		let system_inherent_data = ParachainInherentData {
+			validation_data: vfp.clone(),
+			relay_chain_state,
+			downward_messages: Default::default(),
+			horizontal_messages: Default::default(),
+		};
+		inherent_data
+			.put_data(
+				cumulus_primitives_parachain_inherent::INHERENT_IDENTIFIER,
+				&system_inherent_data,
+			)
+			.expect("failed to put VFP inherent");
+		inherent_data
+	};
+	inherent_data
+}
+
+/// Create contributors.
+fn create_contributors<T: Config>(
+	total_number: u32,
+) -> Vec<(T::RelayChainAccountId, Option<T::AccountId>, BalanceOf<T>)> {
+	let mut contribution_vec = Vec::new();
+	for i in 0..total_number {
+		let seed = SEED - i;
+		let mut account: [u8; 32] = [0u8; 32];
+		let seed_as_slice = seed.to_be_bytes();
+		for j in 0..seed_as_slice.len() {
+			account[j] = seed_as_slice[j]
+		}
+		let relay_chain_account: AccountId32 = account.into();
+		let user = create_funded_user::<T>("user", seed, 0u32.into());
+		let contribution: BalanceOf<T> = 100u32.into();
+		contribution_vec.push((relay_chain_account.into(), Some(user.clone()), contribution));
+	}
+	contribution_vec
+}
+
+/// Insert contributors.
+fn insert_contributors<T: Config>(
+	contributors: Vec<(T::RelayChainAccountId, Option<T::AccountId>, BalanceOf<T>)>,
+) -> Result<(), &'static str> {
+	let mut sub_vec = Vec::new();
+	let batch = max_batch_contributors::<T>();
+
+	for i in 0..contributors.len() {
+		sub_vec.push(contributors[i].clone());
+		if i as u32 % batch == batch - 1 || i == contributors.len() - 1 {
+			Pallet::<T>::initialize_reward_vec(RawOrigin::Root.into(), sub_vec.clone())?;
+			sub_vec.clear()
+		}
+	}
+	Ok(())
+}
+
+/// Create a Contributor.
+fn close_initialization<T: Config>(
+	end_relay: relay_chain::BlockNumber,
+) -> Result<(), &'static str> {
+	Pallet::<T>::complete_initialization(RawOrigin::Root.into(), end_relay)?;
+	Ok(())
+}
+
+fn crate_fake_sig<T: Config>(signed_account: T::AccountId) -> (AccountId32, MultiSignature) {
+	let seed: [u8; 32] = [
+		47, 140, 97, 41, 216, 22, 207, 81, 195, 116, 188, 127, 8, 195, 230, 62, 209, 86, 207, 120,
+		174, 251, 74, 101, 80, 217, 123, 135, 153, 121, 119, 238,
+	];
+	let secret = ed25519_dalek::SecretKey::from_bytes(&seed).unwrap();
+	let public = ed25519_dalek::PublicKey::from(&secret);
+	let pair = ed25519_dalek::Keypair { secret, public };
+	let sig = pair.sign(&signed_account.encode()).to_bytes();
+	let signature: MultiSignature = ed25519::Signature::from_raw(sig).into();
+
+	let ed_public: ed25519::Public = ed25519::Public::unchecked_from(public.to_bytes());
+	let account: AccountId32 = ed_public.into();
+	(account, signature.into())
+}
+
+fn max_batch_contributors<T: Config>() -> u32 {
+	<<T as Config>::MaxInitContributors as Get<u32>>::get()
+}
+
+const MAX_ALREADY_USERS: u32 = 500;
+const MAX_USERS: u32 = 500;
```

Comment:
are these based on the real numbers?

Reply:
We have decided that MAX_ALREADY_USERS will point to the real number of contributors in our crowdloan. MAX_USERS is set to 500 to be able to construct the dependence on it, so not that important that we increase it/decrease it.